src/Encoding.chpl

```diff
@@ -0,0 +1,254 @@
+/* Documentation for Encoding */
+module Encoding {
+
+  use SysCTypes;
+
+  // wireTypes
+  const varint = 0;
+  const lengthDelimited = 2;
+  const fixed64Type = 1;
+  const fixed32Type = 5;
+
+  proc unsignedVarintAppend(val:uint, ch) throws {
+    if val == 0 {
+      ch.write(b"\x00");
+      return;
+    }
+  
+    var newVal = val;
+    var shiftVal: uint;
+    while newVal != 0 {
+      shiftVal = newVal >> 7;
+      var k = if shiftVal != 0 then 0x80 else 0x00;
+      var newByte = (newVal & 0x7F | k):uint(8);
+      ch.write(newByte);
+      newVal = shiftVal;
+    }
+  }
+
+  proc unsignedVarintConsume(ch): (uint, int) throws {
```

Comment:
@mppf I think now we can make the `ch` argument concrete? Something like this ->

`proc unsignedVarintConsume(ch: channel(false,little,false)): (uint, int) throws {`

Reply:
Yes

Reply:
Yup did that :+1: